homeassistant/components/comfoconnect/sensor.py

```diff
@@ -2,93 +2,214 @@
+        ATTR_LABEL: "Exhaust Temperature",
+        ATTR_UNIT: TEMP_CELSIUS,
+        ATTR_ICON: "mdi:thermometer",
+        ATTR_ID: SENSOR_TEMPERATURE_EXHAUST,
+        ATTR_MULTIPLIER: 0.1,
+    },
+    ATTR_EXHAUST_HUMIDITY: {
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_HUMIDITY,
+        ATTR_LABEL: "Exhaust Humidity",
+        ATTR_UNIT: "%",
+        ATTR_ICON: "mdi:water-percent",
+        ATTR_ID: SENSOR_HUMIDITY_EXHAUST,
+    },
+    ATTR_AIR_FLOW_SUPPLY: {
+        ATTR_DEVICE_CLASS: None,
+        ATTR_LABEL: "Supply airflow",
+        ATTR_UNIT: "m³/h",
+        ATTR_ICON: "mdi:fan",
+        ATTR_ID: SENSOR_FAN_SUPPLY_FLOW,
+    },
+    ATTR_AIR_FLOW_EXHAUST: {
+        ATTR_DEVICE_CLASS: None,
+        ATTR_LABEL: "Exhaust airflow",
+        ATTR_UNIT: "m³/h",
+        ATTR_ICON: "mdi:fan",
+        ATTR_ID: SENSOR_FAN_EXHAUST_FLOW,
+    },
+    ATTR_BYPASS_STATE: {
+        ATTR_DEVICE_CLASS: None,
+        ATTR_LABEL: "Bypass State",
+        ATTR_UNIT: "%",
+        ATTR_ICON: "mdi:camera-iris",
+        ATTR_ID: SENSOR_BYPASS_STATE,
+    },
+    ATTR_DAYS_TO_REPLACE_FILTER: {
+        ATTR_DEVICE_CLASS: None,
+        ATTR_LABEL: "Days to replace filter",
+        ATTR_UNIT: "days",
+        ATTR_ICON: "mdi:calendar",
+        ATTR_ID: SENSOR_DAYS_TO_REPLACE_FILTER,
+    },
+    ATTR_POWER_CURRENT: {
+        ATTR_DEVICE_CLASS: DEVICE_CLASS_POWER,
+        ATTR_LABEL: "Power usage",
+        ATTR_UNIT: POWER_WATT,
+        ATTR_ICON: "mdi:flash",
+        ATTR_ID: SENSOR_POWER_CURRENT,
+    },
+}
 
-    global SENSOR_TYPES
-    SENSOR_TYPES = {
-        ATTR_CURRENT_TEMPERATURE: [
-            "Inside Temperature",
-            TEMP_CELSIUS,
-            "mdi:thermometer",
-            SENSOR_TEMPERATURE_EXTRACT,
-        ],
-        ATTR_CURRENT_HUMIDITY: [
-            "Inside Humidity",
-            "%",
-            "mdi:water-percent",
-            SENSOR_HUMIDITY_EXTRACT,
-        ],
-        ATTR_OUTSIDE_TEMPERATURE: [
-            "Outside Temperature",
-            TEMP_CELSIUS,
-            "mdi:thermometer",
-            SENSOR_TEMPERATURE_OUTDOOR,
-        ],
-        ATTR_OUTSIDE_HUMIDITY: [
-            "Outside Humidity",
-            "%",
-            "mdi:water-percent",
-            SENSOR_HUMIDITY_OUTDOOR,
-        ],
-        ATTR_AIR_FLOW_SUPPLY: [
-            "Supply airflow",
-            "m³/h",
-            "mdi:air-conditioner",
-            SENSOR_FAN_SUPPLY_FLOW,
-        ],
-        ATTR_AIR_FLOW_EXHAUST: [
-            "Exhaust airflow",
-            "m³/h",
-            "mdi:air-conditioner",
-            SENSOR_FAN_EXHAUST_FLOW,
-        ],
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_RESOURCES, default=[]): vol.All(
+            cv.ensure_list, [vol.In(SENSOR_TYPES)]
+        ),
     }
+)
+
 
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the ComfoConnect fan platform."""
     ccb = hass.data[DOMAIN]
```

Comment:
When loading via discovery we want a guard clause here that checks if discovery_info is None and return if so.